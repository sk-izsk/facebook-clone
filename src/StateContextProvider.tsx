import React, { createContext, useContext, useReducer } from 'react';

export interface StateContextProviderProps {
  initialState?: any;
  reducer?: any;
}

const StateContext = createContext(undefined);

const StateContextProvider: React.FC<StateContextProviderProps> = ({ initialState, reducer, children }) => {
  return <StateContext.Provider value={useReducer(reducer, initialState) as any}>{children}</StateContext.Provider>;
};

const useStateValue: any = () => useContext(StateContext);

export { StateContextProvider, StateContext, useStateValue };
