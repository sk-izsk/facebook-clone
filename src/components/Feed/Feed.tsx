import { Box, makeStyles } from '@material-ui/core';
import React from 'react';
import { CustomTheme } from '../../theme/muiTheme';

export interface FeedProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  feed: {},
}));

const Feed: React.FC<FeedProps> = () => {
  const classes = useStyles();
  return <Box></Box>;
};

export { Feed };
