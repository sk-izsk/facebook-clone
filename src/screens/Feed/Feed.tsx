import { Box, makeStyles } from '@material-ui/core';
import React, { useEffect, useState } from 'react';
import { fireBaseDb } from '../../api';
import { MessageSender, Post, StoryReel } from '../../components';
import { CustomTheme } from '../../theme/muiTheme';

export interface FeedProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  feed: {
    flex: 1,
    padding: `${theme.spacing(3.75)}px ${theme.spacing(18.75)}px`,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
}));

const Feed: React.FC<FeedProps> = () => {
  const classes = useStyles();
  const [posts, setPosts] = useState([]);

  useEffect(() => {
    fireBaseDb
      .collection('posts')
      .orderBy('timestamp', 'desc')
      .onSnapshot((snapShot: any) => setPosts(snapShot.docs.map((doc: any) => ({ id: doc.id, data: doc.data() }))));
  }, []);

  return (
    <Box className={classes.feed}>
      <StoryReel />
      <MessageSender />
      {posts.map((post: any) => (
        <Post
          key={post.id}
          profilePic={post.data.profilePic}
          message={post.data.message}
          timeStamp={post.data.timestamp}
          userName={post.data.username}
          image={post.data.image}
        />
      ))}
    </Box>
  );
};

export { Feed };
