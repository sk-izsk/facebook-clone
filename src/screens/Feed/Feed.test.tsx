import { render } from '@testing-library/react';
import React from 'react';
import { StateContextProvider } from '../../StateContextProvider';
import { Feed } from './Feed';

const mockUser = {
  image: 'fake image url',
  message: 'fake message',
  profilePic: 'fake profile pic',
  timestamp: 'fake time stamp',
  username: 'fake username',
};

test('Feed renders correctly', () => {
  const { asFragment } = render(
    <StateContextProvider initialState={{ user: mockUser }}>
      <Feed />
    </StateContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
