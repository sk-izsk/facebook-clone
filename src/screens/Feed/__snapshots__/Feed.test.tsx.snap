// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-feed-1"
  >
    <div
      class="MuiBox-root MuiBox-root-4 makeStyles-storyReel-3"
    >
      <div
        class="MuiBox-root MuiBox-root-8 makeStyles-story-5"
        style="background-image: url(https://cdn140.picsart.com/327464379069201.jpg?type=webp&to=crop&r=256);"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-7"
        >
          <img
            class="MuiAvatar-img"
            src="https://cdn131.picsart.com/323779276307201.jpg?type=webp&to=crop&r=256"
          />
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Model
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-story-5"
        style="background-image: url(https://cdn131.picsart.com/315934060092201.jpg?type=webp&to=crop&r=256);"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-7"
        >
          <img
            class="MuiAvatar-img"
            src="https://cdn131.picsart.com/323038550298201.jpg?type=webp&to=crop&r=256"
          />
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Bearded Guy
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-story-5"
        style="background-image: url(https://cdn141.picsart.com/325429489094201.jpg?type=webp&to=crop&r=256);"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-7"
        >
          <img
            class="MuiAvatar-img"
            src="https://cdn131.picsart.com/325488229053201.jpg?type=webp&to=crop&r=256"
          />
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Photoshop
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-11 makeStyles-story-5"
        style="background-image: url(https://images.theconversation.com/files/693/original/aapone-20110118000290280654-file_usa_apple_steve_jobs-original.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=926&fit=clip);"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-7"
        >
          <img
            class="MuiAvatar-img"
            src="https://www.pioneeringminds.com/wp-content/uploads/2018/10/Steve-Jobs-1440x960.jpg"
          />
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Steve Jobs
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-12 makeStyles-story-5"
        style="background-image: url(https://i.pinimg.com/736x/d6/46/8d/d6468d7441b61d051a8c675adbeec09f.jpg);"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-7"
        >
          <img
            class="MuiAvatar-img"
            src="https://img.cinemablend.com/filter:scale/quill/3/3/4/1/1/5/33411540ffe2a4bacfb1824adc71db580b8c7ffd.jpg?mw=600"
          />
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Thanos
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-13 makeStyles-story-5"
        style="background-image: url(https://cdna.artstation.com/p/assets/images/images/017/955/070/large/jackson-caspersz-ironman-snap.jpg?1557977565);"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-7"
        >
          <img
            class="MuiAvatar-img"
            src="https://lc-imageresizer-live-s.legocdn.com/resize/ironman_slider_ironman-photo?width=1600&imageUrl=https%3A%2F%2Fwww.lego.com%2Fr%2Fwww%2Fr%2Fportals%2F-%2Fmedia%2Fcampaigns%2Fgrownup%2Fzebra%2Fironman%2F02-intro-section%2Fironman_slider_ironman-photo.jpg"
          />
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Iron Man
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-14 makeStyles-story-5"
        style="background-image: url(https://images.googlycricket.net/wp-content/uploads/2017/09/Australian-Batsman-Ricky-Ponting-Wallpaper.jpg);"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-7"
        >
          <img
            class="MuiAvatar-img"
            src="https://pbs.twimg.com/profile_images/1204204456092397568/sKIKwz3S_400x400.jpg"
          />
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Ricky Ponting
        </h6>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-24 makeStyles-messageSender-15"
    >
      <div
        class="MuiBox-root MuiBox-root-25 makeStyles-messageSenderTop-16"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <form
          action=""
          class="makeStyles-form-20"
        >
          <input
            class="makeStyles-messageSenderInput-18 makeStyles-inputField-21"
            placeholder="What's on your mind undefined ?"
            type="text"
            value=""
          />
          <input
            class="makeStyles-inputField-21"
            placeholder="Image URL (Optional)"
            type="text"
            value=""
          />
          <button
            class="makeStyles-btn-22"
            type="submit"
          >
            Post
          </button>
        </form>
      </div>
      <div
        class="MuiBox-root MuiBox-root-26 makeStyles-messageSenderBottom-17"
      >
        <div
          class="MuiBox-root MuiBox-root-27 makeStyles-messageSenderOption-19"
        >
          <svg
            color="#ff0000"
            fill="currentColor"
            height="24"
            size="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 0, 0);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 10.5V7c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-text-23 MuiTypography-body1"
          >
            Live video
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-28 makeStyles-messageSenderOption-19"
        >
          <svg
            color="green"
            fill="currentColor"
            height="24"
            size="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: green;"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-text-23 MuiTypography-body1"
          >
            Photo/Video
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-29 makeStyles-messageSenderOption-19"
        >
          <svg
            color="orange"
            fill="currentColor"
            height="24"
            size="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: orange;"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-text-23 MuiTypography-body1"
          >
            Feeling/Activity
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
